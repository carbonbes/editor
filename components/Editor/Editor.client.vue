<template>
  <EditorContent :editor />
</template>

<script lang="ts" setup>
import { EditorContent, type Content } from '@tiptap/vue-3'

const { content } = defineProps<{ content?: Content }>()

const { editor, init, destroy } = useEditor()

onMounted(async () => {
  init(content)
})

onBeforeUnmount(destroy)

useEditorNodesSwipingTracking({
  onSwipeEnd({ xy: [x, y] }) {
    const node = getEditorNodeByCoords(x, y)

    console.log(node)
  },
})
</script>
