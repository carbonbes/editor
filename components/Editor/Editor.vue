<template>
  <EditorContent v-bind="$attrs" />

  <EditorNodeMenuNodesListBottomsheet />
  <EditorNodeMenuNodeActionsListBottomsheet />
  <EditorNodeMenuPopover />
  <EditorSelectionFormattingPopover />
</template>

<script lang="ts" setup>
import type { Content } from '@tiptap/vue-3'

const { content } = defineProps<{ content?: Content }>()

const { init: editorInit, destroy } = useEditor()
const { init: editorCursorTrackingInit } = useEditorNodesCursorTracking()

onMounted(() => {
  editorInit(content)
  editorCursorTrackingInit()
})

onBeforeUnmount(destroy)
</script>
