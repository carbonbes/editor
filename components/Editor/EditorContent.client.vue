<template>
  <EditorContent :editor class="h-full" :class="{ 'overflow-x-hidden': isTouch }" />
</template>

<script lang="ts" setup>
import { EditorContent } from '@tiptap/vue-3'

const { editor } = useEditor()
const { isTouch } = useDevice()
</script>
