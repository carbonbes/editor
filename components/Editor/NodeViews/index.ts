export { default as MediaNodeView } from './MediaNodeView/MediaNodeView.vue'
export { default as YoutubeNodeView } from './YoutubeNodeView/YoutubeNodeView.vue'
