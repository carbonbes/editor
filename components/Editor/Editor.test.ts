import { describe, expect, test } from 'vitest'
import { flushPromises, mount } from '@vue/test-utils'
import Editor from '~/components/Editor/Editor.client.vue'
import type { Content } from '@tiptap/vue-3'

describe('Editor', () => {
  test('should be defined', () => {
    expect(Editor).toBeDefined()
  })

  describe('renders', () => {
    async function mountEditor(content?: Content) {
      const wrapper = mount(Editor, { props: { content } })

      await flushPromises()

      return wrapper
    }

    test('the correct markup without content', async () => {
      const wrapper = await mountEditor()

      expect(wrapper.element.innerHTML).toMatchSnapshot()

      wrapper.unmount()
    })

    test('the correct markup with content', async () => {
      const wrapper = await mountEditor('<h1>Всем привет!</h1>')

      expect(wrapper.element.innerHTML).toMatchSnapshot()

      wrapper.unmount()
    })
  })
})
