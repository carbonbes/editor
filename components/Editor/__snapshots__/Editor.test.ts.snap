// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Editor > renders > the correct markup with content 1`] = `"<div contenteditable="true" role="textbox" translate="no" class="tiptap ProseMirror" tabindex="0"><h1>Всем привет!</h1></div>"`;

exports[`Editor > renders > the correct markup without content 1`] = `"<div contenteditable="true" role="textbox" translate="no" class="tiptap ProseMirror" tabindex="0"><p><br class="ProseMirror-trailingBreak"></p></div>"`;
